src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,326 @@
+        {
+            Cleanup();
+
+            if (RequireNewInstance)
+            {
+                Debug.WriteLine("Starting a new instance of Visual Studio.");
+
+                InitializeHostProcess();
+                InitializeDte();
+                InitializeRemotingService();
+            }
+        }
+
+        internal async Task ExecuteDteCommandAsync(string command, string args = "")
+        {
+            await WaitForDteCommandAvailabilityAsync(command).ConfigureAwait(continueOnCapturedContext: false);
+            _dte.ExecuteCommand(command, args);
+        }
+
+        internal T ExecuteOnHostProcess<T>(Type type, string methodName, BindingFlags bindingFlags, params object[] parameters)
+             => ExecuteOnHostProcess<T>(type.Assembly.Location, type.FullName, methodName, bindingFlags, parameters);
+
+        internal T ExecuteOnHostProcess<T>(string assemblyFilePath, string typeFullName, string methodName, BindingFlags bindingFlags, params object[] parameters)
+        {
+            var objectUri = _service.Execute(assemblyFilePath, typeFullName, methodName, bindingFlags, parameters);
+
+            if (string.IsNullOrWhiteSpace(objectUri))
+            {
+                return default(T);
+            }
+
+            return (T)(Activator.GetObject(typeof(T), $"{_serviceUri}/{objectUri}"));
+        }
+
+        internal async Task<AutomationElement> LocateAutomationElementAsync(string elementName, bool recursive = false)
+        {
+            AutomationElement automationElement = null;
+            var scope = (recursive ? TreeScope.Descendants : TreeScope.Children);
+            var condition = new PropertyCondition(AutomationElement.NameProperty, elementName);
+
+            await IntegrationHelper.WaitForResultAsync(() => {
+                automationElement = AutomationElement.RootElement.FindFirst(scope, condition);
+                return (automationElement != null);
+            }, expectedResult: true).ConfigureAwait(continueOnCapturedContext: false);
+
+            return automationElement;
+        }
+
+        internal Window LocateDteWindow(string windowTitle)
+            => LocateDteWindowAsync(windowTitle).ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
+
+        internal Task<Window> LocateDteWindowAsync(string windowTitle)
+            => IntegrationHelper.WaitForNotNullAsync(() => {
+                foreach (Window window in _dte.Windows)
+                {
+                    if (window.Caption.Equals(windowTitle, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return window;
+                    }
+                }
+                return null;
+            });
+
+        internal async Task WaitForDteCommandAvailabilityAsync(string command)
+            => await IntegrationHelper.WaitForResultAsync(() => Dte.Commands.Item(command).IsAvailable, expectedResult: true).ConfigureAwait(continueOnCapturedContext: false);
```

Comment:
There's no reason for the asynchrony here, just return the underlying task directly.